### PR TITLE
component should be mutable in alpha2

### PR DESCRIPTION
**Pull request by hongchaodeng**

### CODE REVIEW

**Comment by hongchaodeng** [comment]
cc @wonderflow @ryanzhang-oss

File: 7.application_configuration.md [hunk 1]
```diff
@@ -33,7 +33,7 @@ In addition, as an application configuration is released, its _component instanc
 
 To accommodate this definition of a release, an OAM platform SHOULD make the following assumptions:
 
-- A component (as defined in the [component model](4.component.md) section) is immutable. It should not be changed after creation.
+- A component (as defined in the [component model](4.component.md) section) is mutable. Upgrade can happen by updating the component, e.g. updating image.
```
Comment (resouer) [inline 7.application_configuration.md]: Should merge this paragraph with the paragraph below, because change on a Component will results a new release.

Comment (hongchaodeng) [reply 7.application_configuration.md]: Sure. Will do it.

**Comment by negz** [comment]
If we mutate a component any existing workload that uses that component will be updated. Is that really what we want? I don't have a strong opinion either way, but it would be very helpful to explain why this change is being made in the PR.

**Comment by resouer** [comment]
@negz What will be the `Component` upgrade story (let's say upgrade image) in Crossplane? 

My current thought is yes, update Component should trigger rollout of the application by following strategy defined in rollout Trait (if defined).

**Comment by hongchaodeng** [comment]
@negz 

In alpha2, the Component is now mutable. This is a fact and the previous message is outdated.

Rollout is a different story from whether Component is mutable or not. It will involve versions, traffic, rollout strategy, etc. Please get involved in https://github.com/oam-dev/spec/issues/342 and discuss Crossplane use cases there.

**Comment by wonderflow** [comment]
In the view of OAM runtime builder, if component is mutable, then component change should trigger AppConfig Controller to reconcile the application(how to reconcile/rollout is another story).  This may add complexity to OAM runtime, but I think it could work.

In this case we may need to add component version automatically in runtime, like revision in K8s object.

**Comment by resouer** [comment]
Since this issue is closely related to https://github.com/oam-dev/spec/issues/336, and https://github.com/oam-dev/spec/issues/334

We will try to bring up a complete story (upgrade, versioning and immutable) for review this week.

**Comment by ryanzhang-oss** [comment]
Please take a look at https://github.com/oam-dev/spec/issues/336 which is very much related to this.

**Comment by resouer** [comment]
LGTM

**Pull request merged**